ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -144,6 +146,37 @@ def _to_geodataframe(df, schema):
             df = gpd.GeoDataFrame(df, geometry=geom_col)
         return df
 
+    def to_pyarrow_batches(
+        self,
+        expr: ir.Expr,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: str = "default",
+        chunk_size: int = 1_000_000,
+    ) -> Iterable[pa.RecordBatch]:
+        query_ast = self.compiler.to_ast_ensure_limit(
+            expr, limit, params=params
+        )
+        sql = query_ast.compile()
+
+        with self._safe_raw_sql(sql) as cursor:
+            while batch := cursor.fetchmany(chunk_size):
+                struct_array = pa.array(
+                    map(tuple, batch),
+                    type=_schema_to_pyarrow_struct(expr.schema()),
+                )
+                # TODO should we return a RecordBatchReader from this iterator?
+                yield pa.RecordBatch.from_struct_array(struct_array)
+
+    def to_pyarrow(
```

Comment:
For clarity we could call this `to_pyarrow_table`

Reply:
I'm torn on this -- I like things to be explicit AND I don't want to make it too verbose

Reply:
If we consider `pa.Table` as the default pyarrow dataframe-like object, then I'm fine with leaving it as is.

Reply:
Both `cudf` and `polars` call this `to_arrow`, so there's precedent for a simple name. I have a slight preference for `to_pyarrow` over `to_arrow`, but could go either way here. I don't think we need to specify `_table` though.